src/libraries/System.Linq/src/System/Linq/Grouping.SpeedOpt.cs

```diff
@@ -5,51 +5,54 @@
 
 namespace System.Linq
 {
-    internal sealed partial class GroupedResultEnumerable<TSource, TKey, TElement, TResult> : IIListProvider<TResult>
+    public static partial class Enumerable
     {
-        public TResult[] ToArray() =>
-            Lookup<TKey, TElement>.Create(_source, _keySelector, _elementSelector, _comparer).ToArray(_resultSelector);
+        internal sealed partial class GroupByResultIterator<TSource, TKey, TElement, TResult>
```

Comment:
Couldn't these types be marked private now?

Reply:
Yup.